Sources/SwiftGRPC/Core/Channel.swift

```diff
@@ -136,18 +134,16 @@ private extension Channel {
     private let completionQueue: CompletionQueue
     private let underlyingChannel: UnsafeMutableRawPointer
     private let underlyingCompletionQueue: UnsafeMutableRawPointer
-    private let callback: (ConnectivityState) -> Void
-    private var lastState: ConnectivityState
+    private var callbacks = [(ConnectivityState) -> Void]()
     private var hasBeenShutdown = false
-    private let stateMutex: Mutex = Mutex()
+    private let stateMutex = Mutex()
```

Comment:
Not sure if the name `stateMutex` still makes sense, given that it guards `callbacks` and `hasBeenShutdown`.

Reply:
Can rename to simply `mutex`